linked_list.py

```diff
@@ -0,0 +1,348 @@
+
+# Defines a node in the singly linked list
+class Node(object):

```

Comment:
There should be some way to update next_node for a Node. That could be a method like set_next_node